Shopify/tophat
# Prepare Tophat for signed distribution and updates

### What does this change accomplish?

This change updates Tophat so that it can be released as a signed application, and also adds support for automatic updates _via_ [Sparkle](https://github.com/sparkle-project/Sparkle).

Resolves #1

### How have you achieved it?

- Updated the code signing configuration so that the application can be distributed.
- Enabling the Hardened Runtime capability, which is required for Developer ID signing.
- Updated the minimum deployment target to macOS 14 as described in the README.
- Implemented the Sparkle updater in Tophat (will check once a day, prompt to enable automatic updates on second launch).
- Added a GitHub action that automatically signs and produces an `appcast.xml` so that Tophat instances can check for updates.

### How can the change be tested?

Build and run the app, and ensure that any "Check for Updates" controls load briefly and result in an error.  No further testing can be performed as the first `appcast.xml` has not yet been published.
